analyzers/maven/maven_test.go

```diff
@@ -0,0 +1,143 @@
+package maven_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/fossas/fossa-cli/analyzers/maven"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+func TestMavenPOMFile(t *testing.T) {
+	buildTarget := ""
+	m := module.Module{
+		Name:        "test",
+		Type:        pkg.Maven,
+		BuildTarget: buildTarget,
+		Dir:         "test-data",
+	}
+
+	mavenModule := m
+
+	analyzer, err := maven.New(mavenModule)
+	assert.NoError(t, err)
+
+	analyzed, err := analyzer.Analyze()
+	assert.NoError(t, err)
+
+	directDependencies := analyzed.Direct
+
+	compileTimeDependencyfound := false
+	runtTimeDependencyFound := false
+	testDependencyFound := false
+	for dependencyIndex := 0; dependencyIndex < len(directDependencies); dependencyIndex++ {
+		dependencyName := directDependencies[dependencyIndex].Resolved.Name
+		if dependencyName == "com.amazonaws:aws-lambda-java-core" {
+			compileTimeDependencyfound = true
+		}
+		if dependencyName == "com.thoughtworks.xstream:xstream" {
+			runtTimeDependencyFound = true
+		}
+		if dependencyName == "junit:junit" {
+			testDependencyFound = true
+		}
+	}
+
+	assert.Equal(t, compileTimeDependencyfound, true, "Compile time dependency found")
+	assert.Equal(t, runtTimeDependencyFound, true, "Run time dependency found")
+	assert.Equal(t, testDependencyFound, false, "Test dependency not found")
+}
+
+func TestMavenDependencyScope(t *testing.T) {
+	buildTarget := ""
+
+	m := module.Module{
+		Name:        "test",
+		Type:        pkg.Maven,
+		BuildTarget: buildTarget,
+		Dir:         "test-data",
+	}
+
+	// default scope
+	mavenModuleWithDefaultScope := m
+
+	analyzerWithDefaultScope, err := maven.New(mavenModuleWithDefaultScope)
+	assert.NoError(t, err)
+
+	analyzedDefaultScope, err := analyzerWithDefaultScope.Analyze()
+	assert.NoError(t, err)
+
+	// test scope
+	mavenModuleWithTestScope := m
+
+	mavenModuleWithTestScope.Options = map[string]interface{}{
+		"Scope": "test",
+	}
+
+	analyzerWithTestScope, err := maven.New(mavenModuleWithTestScope)
+	assert.NoError(t, err)
+
+	analyzedTestScope, err := analyzerWithTestScope.Analyze()
+	assert.NoError(t, err)
+
+	// runtime scope
+	mavenModuleWithRuntimeScope := m
+
+	mavenModuleWithRuntimeScope.Options = map[string]interface{}{
+		"Scope": "runtime",
+	}
+
+	analyzerWithRuntimeScope, err := maven.New(mavenModuleWithRuntimeScope)
+	assert.NoError(t, err)
+
+	analyzedRuntimeScope, err := analyzerWithRuntimeScope.Analyze()
+	assert.NoError(t, err)
+
+	// analyzedDefaultScope != analyzedTestScope
+	assert.NotEqual(t, analyzedDefaultScope, analyzedTestScope)
+
+	// analyzedDefaultScope == analyzedRuntimeScope
+	assert.Equal(t, analyzedDefaultScope, analyzedRuntimeScope)
```

Comment:
Can you also add a test that the runtime scope contains no test dependencies?

Reply:
```golang
assert.False(t, testDepsExist, "Test dependency not found")
```

Are you talking about this?

Reply:
https://github.com/fossas/fossa-cli/pull/240/files#diff-b9f26dd681f3f09b56b8f03a3f8b2af2R51